src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Authentication;
+
+internal class WebApplicationAuthenticationBuilder : AuthenticationBuilder
+{
+    public bool IsAuthenticationConfigured { get; private set; }
+
+    public WebApplicationAuthenticationBuilder(IServiceCollection services) : base(services) { }
+
+    public override AuthenticationBuilder AddPolicyScheme(string authenticationScheme, string? displayName, Action<PolicySchemeOptions> configureOptions)
+    {
+        RegisterServices(authenticationScheme);
+        return base.AddPolicyScheme(authenticationScheme, displayName, configureOptions);
+    }
+
+    public override AuthenticationBuilder AddRemoteScheme<TOptions, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] THandler>(string authenticationScheme, string? displayName, Action<TOptions>? configureOptions)
+    {
+        RegisterServices(authenticationScheme);
+        return base.AddRemoteScheme<TOptions, THandler>(authenticationScheme, displayName, configureOptions);
+    }
+
+    public override AuthenticationBuilder AddScheme<TOptions, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] THandler>(string authenticationScheme, string? displayName, Action<TOptions>? configureOptions)
+    {
+        RegisterServices(authenticationScheme);
+        return base.AddScheme<TOptions, THandler>(authenticationScheme, displayName, configureOptions);
+    }
+
+    public override AuthenticationBuilder AddScheme<TOptions, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] THandler>(string authenticationScheme, Action<TOptions>? configureOptions)
+    {
+        RegisterServices(authenticationScheme);
+        return base.AddScheme<TOptions, THandler>(authenticationScheme, configureOptions);
+    }
+
+    private void RegisterServices(string authenticationScheme)
+    {
+        IsAuthenticationConfigured = true;
+        Services.AddAuthentication(authenticationScheme);
```

Comment:
There's a bug here, or at least a weirdness, where whatever the last scheme that's added will become the default scheme.  Also override all of the various AddSchemes might not be the best approach either, as this would also miss any schemes that were directly added to the AuthenticationOptions,  I think the safest route to check if any schemes were added is to just get get an `IOptions<AuthenticationOptions>` to look and see if there were any schemes added, and if so, you can add the auth services to the collection and build the service provider again (not sure if there's a fast way to do this?)

Reply:
Or you only call it the first time.
```
if (!IsAuthenticationConfigured)
{
  IsAuthenticationConfigured = true;
  Services.AddAuthentication(authenticationScheme);
  Services.AddAuthorization();
}
```

Reply:
> Or you only call it the first time.

I like this approach since it at least keeps the default scheme at a predictable value.

Added an item to think more on this (and evaluate reading default scheme name from config) in https://github.com/dotnet/aspnetcore/issues/41888.